[CU-ht2xht] Fix creator view

Buglik opened the pull request:
removed green+ button
added tournament name wrapping
added navigation option to the stepper
TODO:
improve validation (if type is null disable button)

Buglik committed:
fixed some bugs
removed green+ button
added tournament name wrapping
added navigation option to the navbar
TODO:
improve validation (if type is null disable button)

Zummek commented:
Task linked: [CU-ht2xht Fix creator view](https://app.clickup.com/t/ht2xht)

Zummek commented on web-app/src/pages/tournament/TournamentCreator.vue:
```diff
@@ -222,6 +225,7 @@ export default class TournamentCreator extends Vue {
   }
 
   private async submitAddTournament() {
+    console.log(this.activeTournamentType);
```
.

Zummek commented on web-app/src/pages/tournament/TournamentCreator.vue:
```diff
@@ -278,4 +282,10 @@ export default class TournamentCreator extends Vue {
 }
+
+.tournamentNameEditor {
+  text-overflow: ellipsis;
```
use `text-overflow` in team list.
![image](https://user-images.githubusercontent.com/24721386/117955483-616c3300-b318-11eb-94d9-f925d66194fd.png)

Zummek replied:
![image](https://user-images.githubusercontent.com/24721386/117955729-a1331a80-b318-11eb-9f57-8d1dbdcc611a.png)

Buglik replied:
![obraz](https://user-images.githubusercontent.com/9595999/118116660-d3ad4800-b3ea-11eb-9bb2-81235eeec8d9.png)
there's a horizontal scroll at desktop view, isn't it better?
doesn't look great but u can actually get the team name :P

Zummek replied:
i suggest to avoid this scroll, add three dots and add tooltip on hover

Zummek reviewed (approved):
Additional matters:
1. src\pages\tournament\TournamentCreator.vue:127 - try to use only one `team-builder` because when we have more than one, we have problem with transferring data between mobile/desktop version
2. 
![image](https://user-images.githubusercontent.com/24721386/117956558-73020a80-b319-11eb-9ffc-c86d956ea40d.png)
This box should have max height. (5 * users.height)
3. i think avatars and/or margins between items in box above should be smaller. We could show more items on this same box height
4. What do you think about remove margin-top from selector?

Anyway code looks good for me

Buglik committed:
added text wrapping

Buglik committed:
added validation
creating blocked if tourntype==null
changed the selectedTournTypeItem view (there was none before)
added tooltip on teamName in TEamsList

Zummek commented on web-app/src/components/tournament/creator/TeamBuilder.vue:
```diff
@@ -212,3 +216,9 @@ export default class TeamBuilder extends Vue {
   }
 }
 </script>
+<style scoped>
+.playerSelectScope {
+  padding-top: 0;
+  padding-bottom: 0;
+}
+</style>
```
You can replace this to quasar class q-py-none

Zummek commented on web-app/src/components/tournament/creator/TeamsList.vue:
```diff
@@ -41,8 +41,11 @@
             />
           </template>
-          <template v-else>
+          <template v-if="col.name === 'name'">
```
Use :key props instead of condition for assign template for column

Zummek commented on web-app/src/components/tournament/creator/TournamentTypeSelector.vue:
```diff
@@ -5,18 +5,23 @@
   components: { SelectableTournamentType },
 })
 export default class TournamentTypeSelector extends Vue {
+  @Prop({ type: String || null, required: true })
+  activeType!: TournamentType | null;
```
Is this prop nullable? I think it can be TournamenType only, when its show error try to assign one of value., then overwrite it.

Zummek replied:
Could this prop be boolean? And tell me a different between `activeType` and `tournamentType` props

Buglik replied:
it is null before choosing tourn type

``activeType`` is the type that is currently selected ``tournamentType`` whereas is the one u can select 
if the ``activeType`` and ``tournamentType`` of element are equal it means component should have "selected" behaviour

Zummek commented on web-app/src/pages/tournament/TournamentCreator.vue:
```diff
@@ -14,16 +14,16 @@
         :title="$t('tournament.stepper.buildTeams')"
         icon="settings"
-        :error="(activeTournamentType = null)"
+        :error="step > 2 && teams.length < 2"
         :done="step > 2"
```
I'm not sure
error prop is true when step is grater than 2 and&&... and done prop is true when step is greate than 2 too.
Is it correct?

Buglik replied:
this will be changed after adding 3rd step (scheduling)

Zummek commented on web-app/src/pages/tournament/TournamentCreator.vue:
```diff
@@ -191,18 +191,22 @@ export default class TournamentCreator extends Vue {
       align: 'left',
       field: 'name',
       sortable: true,
+      classes: 'textWrapDotted',
```
`classes` name refer to array of class names/strings, not to for single/standard string

Buglik committed:
changed table cols templates assignment

Zummek commented on web-app/src/components/tournament/creator/TeamBuilder.vue:
```diff
@@ -216,9 +216,4 @@ export default class TeamBuilder extends Vue {
-  padding-bottom: 0;
-}
-</style>
+<style scoped></style>
```
```suggestion
```

Zummek commented on web-app/src/pages/tournament/TournamentCreator.vue:
```diff
@@ -191,22 +191,21 @@ export default class TournamentCreator extends Vue {
     },
     {
       name: 'participants',
+      required: true,
```
Is `required` is  required? 
In doc `required` has desc "If we use visible-columns, this col will always be visible".
I think, You are not using `visible-columns`

Buglik committed:
changes after CR
